BUG: `check_dtype=False` in `assert_series_equal` is not returning expected results for `datetime` & `timedelta` types in `pandas-2.0`

### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [ ] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.


### Reproducible Example

```python
In [1]: import pandas as pd

In [2]: s = pd.Series([1000213, 2131232, 21312331], dtype='datetime64[s]')

In [3]: s
Out[3]: 
0   1970-01-12 13:50:13
1   1970-01-25 16:00:32
2   1970-09-04 16:05:31
dtype: datetime64[s]

In [4]: p = s.astype('datetime64[ms]')

In [6]: p
Out[6]: 
0   1970-01-12 13:50:13
1   1970-01-25 16:00:32
2   1970-09-04 16:05:31
dtype: datetime64[ms]

In [7]: s
Out[7]: 
0   1970-01-12 13:50:13
1   1970-01-25 16:00:32
2   1970-09-04 16:05:31
dtype: datetime64[s]

In [8]: pd.testing.assert_series_equal(s, p)          # Failure: Works as expected since `dtype's` are different
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
Cell In[8], line 1
----> 1 pd.testing.assert_series_equal(s, p)

    [... skipping hidden 2 frame]

File /nvme/0/pgali/envs/cudfdev/lib/python3.10/site-packages/pandas/_testing/asserters.py:596, in raise_assert_detail(obj, message, left, right, diff, first_diff, index_values)
    593 if first_diff is not None:
    594     msg += f"\n{first_diff}"
--> 596 raise AssertionError(msg)

AssertionError: Attributes of Series are different

Attribute "dtype" are different
[left]:  datetime64[s]
[right]: datetime64[ms]

In [9]: pd.testing.assert_series_equal(s, p, check_dtype=False)       # I expect this to not raise, because we are asking for the dtypes to be ignored and the data as seen above is perfectly identical.
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
Cell In[9], line 1
----> 1 pd.testing.assert_series_equal(s, p, check_dtype=False)

    [... skipping hidden 1 frame]

File /nvme/0/pgali/envs/cudfdev/lib/python3.10/site-packages/pandas/_testing/asserters.py:741, in assert_extension_array_equal(left, right, check_dtype, index_values, check_exact, rtol, atol, obj)
    732     assert_attr_equal("dtype", left, right, obj=f"Attributes of {obj}")
    734 if (
    735     isinstance(left, DatetimeLikeArrayMixin)
    736     and isinstance(right, DatetimeLikeArrayMixin)
   (...)
    739     # Avoid slow object-dtype comparisons
    740     # np.asarray for case where we have a np.MaskedArray
--> 741     assert_numpy_array_equal(
    742         np.asarray(left.asi8),
    743         np.asarray(right.asi8),
    744         index_values=index_values,
    745         obj=obj,
    746     )
    747     return
    749 left_na = np.asarray(left.isna())

    [... skipping hidden 1 frame]

File /nvme/0/pgali/envs/cudfdev/lib/python3.10/site-packages/pandas/_testing/asserters.py:666, in assert_numpy_array_equal.<locals>._raise(left, right, err_msg)
    664     diff = diff * 100.0 / left.size
    665     msg = f"{obj} values are different ({np.round(diff, 5)} %)"
--> 666     raise_assert_detail(obj, msg, left, right, index_values=index_values)
    668 raise AssertionError(err_msg)

File /nvme/0/pgali/envs/cudfdev/lib/python3.10/site-packages/pandas/_testing/asserters.py:596, in raise_assert_detail(obj, message, left, right, diff, first_diff, index_values)
    593 if first_diff is not None:
    594     msg += f"\n{first_diff}"
--> 596 raise AssertionError(msg)

AssertionError: Series are different

Series values are different (100.0 %)
[index]: [0, 1, 2]
[left]:  [1000213, 2131232, 21312331]
[right]: [1000213000, 2131232000, 21312331000]
```


### Issue Description

With the newly introduced `datetime64` & `timedelta64` time resolutions, it is possible to hold the identical data in different dtypes. So when we pass `check_dtype=False` to `assert_frame_equal` we expect identical data to pass and not raise an error.

### Expected Behavior

```python
In [9]: pd.testing.assert_series_equal(s, p, check_dtype=False) # Passes.

In [10]: pd.testing.assert_series_equal(s, p, check_dtype=True) # Raises error
```

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : c2a7f1ae753737e589617ebaaff673070036d653
python           : 3.10.10.final.0
python-bits      : 64
OS               : Linux
OS-release       : 4.15.0-76-generic
Version          : #86-Ubuntu SMP Fri Jan 17 17:24:28 UTC 2020
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 2.0.0rc1
numpy            : 1.23.5
pytz             : 2023.3
dateutil         : 2.8.2
setuptools       : 67.6.1
pip              : 23.0.1
Cython           : 0.29.33
pytest           : 7.2.2
hypothesis       : 6.70.1
sphinx           : 5.3.0
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 3.1.2
IPython          : 8.12.0
pandas_datareader: None
bs4              : 4.12.0
bottleneck       : None
brotli           : 
fastparquet      : None
fsspec           : 2023.3.0
gcsfs            : None
matplotlib       : None
numba            : 0.56.4
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : 11.0.0
pyreadstat       : None
pyxlsb           : None
s3fs             : 2023.3.0
scipy            : 1.10.1
snappy           : 
sqlalchemy       : 1.4.46
tables           : None
tabulate         : 0.9.0
xarray           : None
xlrd             : None
zstandard        : None
tzdata           : 2023.3
qtpy             : None
pyqt5            : None

</details>

--- a/doc/source/whatsnew/v2.0.1.rst
+++ b/doc/source/whatsnew/v2.0.1.rst
@@ -20,7 +20,7 @@ Fixed regressions
 
 Bug fixes
 ~~~~~~~~~
--
+- Bug in :func:`pandas.testing.assert_series_equal` where ``check_dtype=False`` would still raise for datetime or timedelta types with different resolutions (:issue:`52449`)
 
 .. ---------------------------------------------------------------------------
 .. _whatsnew_201.other:

--- a/pandas/_testing/asserters.py
+++ b/pandas/_testing/asserters.py
@@ -1,5 +1,6 @@
 from __future__ import annotations
 
+import operator
 from typing import (
     TYPE_CHECKING,
     Literal,
@@ -11,6 +12,10 @@ import numpy as np
 from pandas._libs.missing import is_matching_na
 from pandas._libs.sparse import SparseIndex
 import pandas._libs.testing as _testing
+from pandas._libs.tslibs.np_datetime import (
+    compare_mismatched_resolutions,
+    py_get_unit_from_dtype,
+)
 
 from pandas.core.dtypes.common import (
     is_bool,
@@ -744,6 +749,17 @@ def assert_extension_array_equal(
         and isinstance(right, DatetimeLikeArrayMixin)
         and type(right) == type(left)
     ):
+        # GH 52449
+        if (
+            not check_dtype
+            and left.dtype.kind in "mM"
+            and py_get_unit_from_dtype(left.dtype)
+            != py_get_unit_from_dtype(right.dtype)
+        ):
+            if compare_mismatched_resolutions(
+                left._ndarray, right._ndarray, operator.eq
+            ).all():
+                return
         # Avoid slow object-dtype comparisons
         # np.asarray for case where we have a np.MaskedArray
         assert_numpy_array_equal(

--- a/pandas/_testing/asserters.py
+++ b/pandas/_testing/asserters.py
@@ -12,10 +12,7 @@ import numpy as np
 from pandas._libs.missing import is_matching_na
 from pandas._libs.sparse import SparseIndex
 import pandas._libs.testing as _testing
-from pandas._libs.tslibs.np_datetime import (
-    compare_mismatched_resolutions,
-    py_get_unit_from_dtype,
-)
+from pandas._libs.tslibs.np_datetime import compare_mismatched_resolutions
 
 from pandas.core.dtypes.common import (
     is_bool,
@@ -750,15 +747,21 @@ def assert_extension_array_equal(
         and type(right) == type(left)
     ):
         # GH 52449
-        if (
-            not check_dtype
-            and left.dtype.kind in "mM"
-            and py_get_unit_from_dtype(left.dtype)
-            != py_get_unit_from_dtype(right.dtype)
-        ):
-            if compare_mismatched_resolutions(
-                left._ndarray, right._ndarray, operator.eq
-            ).all():
+        if not check_dtype and left.dtype.kind in "mM":
+            if not isinstance(left.dtype, np.dtype):
+                l_unit = left.dtype.unit
+            else:
+                l_unit = np.datetime_data(left.dtype)[0]
+            if not isinstance(right.dtype, np.dtype):
+                r_unit = right.dtype.unit
+            else:
+                r_unit = np.datetime_data(right.dtype)[0]
+            if (
+                l_unit != r_unit
+                and compare_mismatched_resolutions(
+                    left._ndarray, right._ndarray, operator.eq
+                ).all()
+            ):
                 return
         # Avoid slow object-dtype comparisons
         # np.asarray for case where we have a np.MaskedArray

--- a/pandas/_testing/asserters.py
+++ b/pandas/_testing/asserters.py
@@ -750,13 +750,11 @@ def assert_extension_array_equal(
         # GH 52449
         if not check_dtype and left.dtype.kind in "mM":
             if not isinstance(left.dtype, np.dtype):
-                left = cast(DatetimeTZDtype, left)
-                l_unit = left.dtype.unit
+                l_unit = cast(DatetimeTZDtype, left.dtype).unit
             else:
                 l_unit = np.datetime_data(left.dtype)[0]
             if not isinstance(right.dtype, np.dtype):
-                right = cast(DatetimeTZDtype, right)
-                r_unit = right.dtype.unit
+                r_unit = cast(DatetimeTZDtype, left.dtype).unit
             else:
                 r_unit = np.datetime_data(right.dtype)[0]
             if (

--- a/pandas/_testing/asserters.py
+++ b/pandas/_testing/asserters.py
@@ -24,6 +24,7 @@ from pandas.core.dtypes.common import (
 )
 from pandas.core.dtypes.dtypes import (
     CategoricalDtype,
+    DatetimeTZDtype,
     ExtensionDtype,
     PandasDtype,
 )
@@ -749,10 +750,12 @@ def assert_extension_array_equal(
         # GH 52449
         if not check_dtype and left.dtype.kind in "mM":
             if not isinstance(left.dtype, np.dtype):
+                left = cast(DatetimeTZDtype, left)
                 l_unit = left.dtype.unit
             else:
                 l_unit = np.datetime_data(left.dtype)[0]
             if not isinstance(right.dtype, np.dtype):
+                right = cast(DatetimeTZDtype, right)
                 r_unit = right.dtype.unit
             else:
                 r_unit = np.datetime_data(right.dtype)[0]

--- a/pandas/tests/util/test_assert_series_equal.py
+++ b/pandas/tests/util/test_assert_series_equal.py
@@ -405,3 +405,21 @@ def test_identical_nested_series_is_equal():
     tm.assert_series_equal(x, x, check_exact=True)
     tm.assert_series_equal(x, y)
     tm.assert_series_equal(x, y, check_exact=True)
+
+
+@pytest.mark.parametrize("dtype", ["datetime64", "timedelta64"])
+def test_check_dtype_false_different_reso(dtype):
+    # GH 52449
+    ser_s = Series([1000213, 2131232, 21312331]).astype(f"{dtype}[s]")
+    ser_ms = ser_s.astype(f"{dtype}[ms]")
+    with pytest.raises(AssertionError, match="Attributes of Series are different"):
+        tm.assert_series_equal(ser_s, ser_ms)
+    tm.assert_series_equal(ser_ms, ser_s, check_dtype=False)
+
+    ser_ms -= Series([1, 1, 1]).astype(f"{dtype}[ms]")
+
+    with pytest.raises(AssertionError, match="Series are different"):
+        tm.assert_series_equal(ser_s, ser_ms)
+
+    with pytest.raises(AssertionError, match="Series are different"):
+        tm.assert_series_equal(ser_s, ser_ms, check_dtype=False)

